Adding time-varying methods, under-ascertainment methods and vignettes for all of the above (#23)
epiverse-trace/datadelay: vignettes/estimate_ascertainment.Rmd

---
title: "Estimating the proportion of cases that are reported during an outbreak"
output: rmarkdown::html_vignette
bibliography: resources/library.bib
csl: resources/bmj.csl
vignette: >
  %\VignetteIndexEntry{Estimating the proportion of cases that are reported during an outbreak}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

<!-- removes the ugly border around the figures -->
<style>
    img {
        border: 0;
    }
</style>

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.align = "left",
  fig.width = 600 / 80,
  fig.height = 400 / 80,
  dpi = 80
)
```

# Overview

This vignette outlines the function within `datadelay` used to estimate the
proportion of cases, infections or hospitalisations ascertained.

> We load the three packages required in this vignette using the following 
> commands:
> 
> Now we load the five key packages we need to run the commands in this vignette:

Reviewer: ```suggestion
First, we load the key packages required for this vignette:
```

```{r, message = FALSE, warning=FALSE, eval = TRUE}
library(datadelay)
library(epiparameter)
library(covidregionaldata)
library(knitr)
library(kableExtra)
```

and the four optional packages, to run the multiple examples in the last 
section of this vignette:

```{r, message = FALSE, warning=FALSE, eval = TRUE}
library(dplyr)
library(ggplot2)
library(forcats)
library(scales)
```

# Motivation

There are several reasons why cases, infections or hospitalisations and even 
deaths might be under-ascertained during an outbreak of an infectious disease. 
Some examples include:

* Testing effort or ability is not high enough to catch all infections.

* Testing triggered by individuals rather than sampling at random in the 
population. This can lead to biases due to some infections either being 
asymptomatic or having a pre-symptomatic phase. For example, if testing is 
initiated by an individual who appears not to have any symptoms, it is likely 
the ascertainment of their infection will be delayed, or possibly missed 
entirely.

> * Changes in the criteria, either used to diagnose the disease in question or
> to determine whether a death can be atttributed to the disease in question.

Reviewer: ```suggestion
* Changes in case definition over time.
```

> For example, during the recent COVID-19 pandemic, the U.K. were not able to roll

Reviewer: ```suggestion
For example, during the recent COVID-19 pandemic, the U.K. did not roll
```

> out widespread testing until around May 2020. This means that during March and 
> April, infections were significantly under-ascertained. We focus on this period

Reviewer: ```suggestion
out widespread testing until around May 2020. This means that during March and 
April, infections were heavily under-ascertained. We focus on this period
```

command:

```{r message=FALSE, warning=FALSE, eval=TRUE}
onset_to_death_covid <- epidist_db(
  disease = "COVID-19",
  epi_dist = "onset_to_death",
  author = "Linton_etal"
)
```

To visualise this distribution, we evaluate it between 0 and 30 days, and plot
the results over time. We do so using the following command:

```{r message=FALSE, warning=FALSE, eval=TRUE, fig.cap = "\\label{fig:delay_dist_covid} **Figure 3: Example plot of the appropriate delay distribution for the COVID-19 epidemic in the U.K.** We plot the onset-to-death distribution we use throughout this example for COVID-19, reported in https://doi.org/10.3390/jcm9020538."}
plot_epiparameter_distribution(onset_to_death_covid, by = 0.1)
```

## Estimating the proportion of cases that have been ascertained

We use the `estimate_reporting()` function within the `datadelay` package to 
calculate the time-varying CFR for the COVID-19 epidemic in the U.K:

```{r message = FALSE, warning = FALSE, eval = TRUE}
df_reporting_static <- estimate_reporting(df_covid_uk_subset,
  epi_dist = onset_to_death_covid,
  type = "static",
  severity_baseline = 0.014,
  correct_for_delays = TRUE
) |>

>   format_output(estimate_type = "reporting", type = "Under-reporting")

Reviewer: "Percent ascertained" might be clearer than "under-reporting", if type is easily changeable.

Reviewer 2: Now converted to issue #26
